async-clipboard/demo.js

```diff
@@ -0,0 +1,55 @@
+const copy = document.querySelector('#copy');
+const paste = document.querySelector('#paste');
+const out = document.querySelector('#out');
+
+/** Write contents of the textarea to the clipboard when clicking "Copy" */
+copy.onclick = () => {
+  navigator.clipboard.writeText(out.value)
+    .then(result => {
+      ChromeSamples.log('Text copied.' + result);
+    })
+    .catch(err => {
+      ChromeSamples.log('Failed to copy text: ' + err);
+    });
+};
+
+/** Read from clipboard when clicking the Paste button */
+paste.onclick = () => {
+  navigator.clipboard.readText()
+    .then(text => {
+      out.value = text;
+      ChromeSamples.log('Text pasted.');
+    })
+    .catch(err => {
+      ChromeSamples.log('Failed to read from clipboard: ' + err);
+    });
+};
+
+/** Watch for pastes */
+document.addEventListener('paste', e => {
+  e.preventDefault();
+  navigator.clipboard.getText().then(text => {
```

Comment:
I get an error there as well: `Uncaught TypeError: navigator.clipboard.getText is not a function`.

Reply:
Are you running it on a secure origin or localhost?